src/overlay/test/ItemFetcherTests.cpp

```diff
@@ -221,41 +223,125 @@ TEST_CASE("ItemFetcher fetches", "[overlay][ItemFetcher]")
 
             LoopbackPeerConnection connection2(*app, *other2);
             auto peer2 = connection2.getInitiator();
-            SECTION("fetching once works")
+
+            SECTION("success")
             {
-                auto zeroEnvelope1 = makeEnvelope(0);
-                itemFetcher.fetch(zero, zeroEnvelope1);
-            }
-            SECTION("fetching twice does not trigger any additional network "
+                SECTION("fetching once works")
+                {
+                    auto zeroEnvelope1 = makeEnvelope(0);
+                    itemFetcher.fetch(zero, zeroEnvelope1);
+                }
+                SECTION(
+                    "fetching twice does not trigger any additional network "
                     "activity")
+                {
+                    auto zeroEnvelope1 = makeEnvelope(0);
+                    auto zeroEnvelope2 = makeEnvelope(0);
+                    itemFetcher.fetch(zero, zeroEnvelope1);
+                    itemFetcher.fetch(zero, zeroEnvelope2);
+                }
+                REQUIRE(asked.size() == 0);
+
+                while (asked.size() < 4)
+                {
+                    clock.crank(true);
+                }
+
+                itemFetcher.recv(zero, timer);
+
+                // crank for a while
+                constexpr auto elapsed = std::chrono::minutes(1);
+                auto const later = clock.now() + elapsed;
+                while (clock.crank(false) > 0 && clock.now() < later)
+                {
+                    clock.sleep_for(std::chrono::milliseconds(500));
+                }
+
+                REQUIRE(asked.size() == 4);
+
+                REQUIRE(std::count(asked.begin(), asked.end(), peer1) == 2);
+                REQUIRE(std::count(asked.begin(), asked.end(), peer2) == 2);
+            }
+            SECTION("not found")
             {
                 auto zeroEnvelope1 = makeEnvelope(0);
-                auto zeroEnvelope2 = makeEnvelope(0);
                 itemFetcher.fetch(zero, zeroEnvelope1);
-                itemFetcher.fetch(zero, zeroEnvelope2);
-            }
-            REQUIRE(asked.size() == 0);
 
-            while (asked.size() < 4)
-            {
-                clock.crank(true);
+                std::chrono::milliseconds const FETCH_NEXT_PEER{1500};
+
+                auto testNotFound = [&](bool respond) {
+                    int constexpr ITERATIONS = 100;
+                    for (auto i = ITERATIONS; i != 0; --i)
```

Comment:
is this loop to test what happens when we run out of peers? curious why we need so many iterations here